src/main/webapp/app/entities/oauth-client/oauth-client-dialog.component.html

```diff
@@ -0,0 +1,75 @@
+<form name="editForm" role="form" novalidate (ngSubmit)="save()" #editForm="ngForm">
+
+    <div class="modal-header">
+        <h4 class="modal-title" id="myOAuthClientLabel" jhiTranslate="managementPortalApp.oauthClient.home.createOrEditLabel">Create or edit an OAuth Client</h4>
+        <button type="button" class="close" data-dismiss="modal" aria-hidden="true"
+                (click)="clear()">&times;</button>
+    </div>
+    <div class="modal-body">
+        <jhi-alert-error></jhi-alert-error>
+        <div class="alert alert-danger" *ngIf="protectedClient">
+            <strong>This is a protected client.</strong> You will not be able to make any changes to it.
+        </div>
+        <div class="form-group">
+            <label class="form-control-label" for="id" jhiTranslate="managementPortalApp.oauthClient.clientId">Client ID</label>
+            <input *ngIf="!newClient" type="text" class="form-control" id="id" name="id" [(ngModel)]="client.clientId" readonly />
+            <input *ngIf="newClient" type="text" class="form-control" id="id" name="id" [(ngModel)]="client.clientId" required />
+        </div>
+        <div class="form-group">
+            <label class="form-control-label" for="" jhiTranslate="managementPortalApp.oauthClient.clientSecret">Client Secret</label>
```

Comment:
Not sure whether its possible. Can we change the values into `password` mode when the user switches to other UI element or have a hide button?